File: C#/Searching/Searching.cs
using System;
using System.IO;
using System.Net;
using System.Collections.Generic;

// Newtonsoft.Json is available from http://www.newtonsoft.com/json
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

public class Searching
{
    static public void Main()
    {
        string baseURL = "{base url}/rest/latest/";
        
        // Username and password should be stored according
        // to your organization's security policies
        string username = "API_User";
        string password = "********";

        string stringToFind = "Unique string";

        string containsParameter = "contains=" + WebUtility.UrlEncode(stringToFind);

        string url = baseURL + "abstractitems?" + containsParameter;

        HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
        request.Credentials = new NetworkCredential(username, password);
        request.Method = WebRequestMethods.Http.Get;
        request.PreAuthenticate = true;

        HttpWebResponse response = (HttpWebResponse)request.GetResponse();

        Stream stream = response.GetResponseStream();
        StreamReader streamReader = new StreamReader(stream);

        JObject jsonResponse = JObject.Parse(streamReader.ReadToEnd());
        JObject meta = (JObject) jsonResponse["meta"];
        JObject pageInfo = (JObject) meta["pageInfo"];
        long totalResults = (long) pageInfo["totalResults"];

        if (totalResults == 1)
        {
            JArray data = (JArray) jsonResponse["data"];
            JObject item = (JObject) data[0];

            JObject fields = (JObject) item["fields"];
            string name = (string) fields["name"];
            Console.WriteLine(name);
        }
        else
        {
            Console.WriteLine("stringToFind wasn't unique");
        }
    }
}

File: Ruby/Searching/searching.rb
require "net/http"
require "json"
require "uri"

base_url = "{base url}/rest/latest/"

# Username and password should be stored according
# to your organization's security policies
username = "API_User"
password = "********"

string_to_find = "Unique string"

contains_parameter = URI.escape("contains=#{string_to_find}")

uri = URI("#{base_url}abstractitems?#{contains_parameter}")
Net::HTTP.start(uri.host, uri.port, 
    :use_ssl => uri.scheme == 'https') do |http| 
    request = Net::HTTP::Get.new(uri)
    request.basic_auth(username, password)
    response = http.request(request)
    
    json_response = JSON.parse(response.body)
    total_results = json_response["meta"]["pageInfo"]["totalResults"]

    if total_results == 1 then
        data = json_response["data"]
        item = data[0]

        puts item["fields"]["name"]
    else
        puts "string_to_find wasn't unique"
    end
end
